subxt/src/storage/storage_type.rs

```diff
@@ -237,6 +241,43 @@ where
             })
         }
     }
+
+    /// the storage version of a pallet.
+    pub async fn storage_version(
+        &self,
+        pallet_name: impl AsRef<str>,
+    ) -> Result<StorageVersion, Error> {
+        // check that the pallet exists in the metadata:
+        self.client
+            .metadata()
+            .pallet_by_name(pallet_name.as_ref())
+            .ok_or_else(|| MetadataError::PalletNameNotFound(pallet_name.as_ref().into()))?;
+
+        // construct the storage key. This is done similarly in `frame_support::traits::metadata::StorageVersion::storage_key()`.
+        pub const STORAGE_VERSION_STORAGE_KEY_POSTFIX: &[u8] = b":__STORAGE_VERSION__:";
+        let mut key_bytes: Vec<u8> = vec![];
+        key_bytes.extend(&sp_core_hashing::twox_128(pallet_name.as_ref().as_bytes()));
+        key_bytes.extend(&sp_core_hashing::twox_128(
+            STORAGE_VERSION_STORAGE_KEY_POSTFIX,
+        ));
+
+        // fetch the raw bytes and decode them into the StorageVersion struct:
+        let storage_version_bytes = self.fetch_raw(&key_bytes).await?.ok_or(format!(
```

Comment:
Nit: `ok_or_else` here toa void formatting the string unless there's an error (I'd expect clippy to spot this but havent checked)